blockchain/src/blockchain/history_sync.rs

```diff
@@ -273,7 +115,7 @@ impl Blockchain {
             if policy::is_macro_block_at(*block_number) {
                 let finalize_batch = Inherent {
                     ty: InherentType::FinalizeBatch,
-                    target: this.staking_contract_address(),
+                    target: stacking_contract_address.clone(),
```

Comment:
```suggestion
                    target: staking_contract_address.clone(),
```

Reply:
There's still a test called `it_can_revert_create_stacker_transaction` (which I think is not from your changeset, but while you're at it we can also rename that 😄 )